txindex.go

```diff
@@ -57,27 +57,30 @@ func (lw *LibWallet) IndexTransactions(afterIndexing func()) error {
 		}
 	}
 
-	beginHeight, err := lw.txDB.ReadIndexingStartBlock()
+	beginHeight, err := wallet.txDB.ReadIndexingStartBlock()
 	if err != nil {
-		log.Errorf("Get tx indexing start point error: %v", err)
+		log.Errorf("[%d] Get tx indexing start point error: %v", wallet.ID, err)
 		return err
 	}
 
-	endHeight := lw.GetBestBlock()
+	endHeight := wallet.GetBestBlock()
 
-	startBlock := wallet.NewBlockIdentifierFromHeight(beginHeight)
-	endBlock := wallet.NewBlockIdentifierFromHeight(endHeight)
+	startBlock := w.NewBlockIdentifierFromHeight(beginHeight)
+	endBlock := w.NewBlockIdentifierFromHeight(endHeight)
 
 	defer func() {
-		afterIndexing()
-		count, err := lw.txDB.Count(txindex.TxFilterAll, &Transaction{})
+		count, err := wallet.txDB.Count(txindex.TxFilterAll, &Transaction{})
 		if err != nil {
-			log.Errorf("Post-indexing tx count error :%v", err)
-			return
+			log.Errorf("[%d] Post-indexing tx count error :%v", wallet.ID, err)
+		} else if count > 0 {
+			log.Debugf("[%d] Transaction index finished at %d, %d transaction(s) indexed in total", wallet.ID, txEndHeight, count)
 		}
-		log.Infof("Transaction index finished at %d, %d transaction(s) indexed in total", txEndHeight, count)
+
+		waitGroup.Done()
 	}()
 
-	log.Infof("Indexing transactions start height: %d, end height: %d", beginHeight, endHeight)
-	return lw.wallet.GetTransactions(rangeFn, startBlock, endBlock)
+	waitGroup.Add(1)
+	log.Debugf("[%d] Indexing transactions start height: %d, end height: %d", wallet.ID, beginHeight, endHeight)
+	wallet.internal.GetTransactions(ctx, rangeFn, startBlock, endBlock)
```

Comment:
Should not ignore the error returned here:
```suggestion
	return wallet.internal.GetTransactions(ctx, rangeFn, startBlock, endBlock)
```